backend-python/src/telestion/backend/lib.py

```diff
@@ -0,0 +1,162 @@
+        return await self.nc.subscribe(**kwargs)
+
+    async def request(self, **kwargs) -> NatsMsg:
+        """Wrapper for Client.request(subject, payload, timeout, old_style, headers)"""
+        return await self.nc.request(**kwargs)
+
+    async def flush(self, timeout: int = DEFAULT_FLUSH_TIMEOUT) -> None:
+        """Wrapper for https://nats-io.github.io/nats.py/modules.html#nats.aio.client.Client.flush"""
+        await self.nc.flush(timeout)
+
+    async def drain(self) -> None:
+        """Wrapper for https://nats-io.github.io/nats.py/modules.html#nats.aio.client.Client.drain"""
+        await self.nc.drain()
+
+    async def close(self) -> None:
+        """Wrapper for https://nats-io.github.io/nats.py/modules.html#nats.aio.client.Client.close"""
+        await self.nc.close()
+
+
+@dataclass
+class Options:
+    """Storing a custom configuration which overwrites the parsed config during startup of the Telestion service."""
+    nats: bool = True
+    """Whether a service should use nats. If set to False no nats client is set up during startup"""
+    # (officially) we don't support int keys, btw...
+    overwrite_args: dict[str, Any] | None = None
+    """Arguments overwriting the parsed config of a service."""
+    custom_nc: NatsClient | None = None
+    """Custom nats client. During startup no configuration takes place if present."""
+
+    def without_nats(self) -> 'Options':
+        """Returns a copy of this Options instance with nats switched off."""
+        return replace(self, nats=False, custom_nc=None)
+
+    def with_overwrite_args(self, **kwargs) -> 'Options':
+        """Returns a copy of this Options instance with different custom arguments."""
+        return replace(self, overwrite_args=kwargs)
+
+    def with_custom_nc(self, nats_client: NatsClient) -> 'Options':
+        """Returns a copy of this Options instance with a custom client."""
+        return replace(self, custom_nc=nats_client)
+
+
+async def setup_healthcheck(nc: NatsClient, service_name: str) -> NatsClient:
+    """Sets up __telestion__.health for a NatsClient. Returns NatsClient for fluent API."""
+    async def _respond_hc(msg):
+        msg.respond(
+            json_encode({
+                "errors": 0,
+                "name": service_name
+            })
+        )
+
+    await nc.subscribe(
+        '__telestion__.health',
+        cb=_respond_hc
+    )
+
+    return nc
+
+
+async def start_service(opts: Options = None) -> Service:
+    """Creates a Service for the given Options and the parsed config and spins up a new NATS client if configured so."""
+    if opts is None:
+        opts = Options()
+
+    config = build_config()
+    if opts.overwrite_args is not None:
+        config = config.model_copy(update=opts.overwrite_args)
+
+    service = Service(opts.custom_nc, config.data_dir, config.service_name, config)
+
+    if not opts.nats or opts.custom_nc is not None:
+        return service
```

Comment:
Also custom NATS proxies / mocks can connect => this differentiation is unneeded. Quite the opposite: you still need the health check setup to enable testing of that feature.